packages/application/style/menus.css

```diff
@@ -38,6 +38,24 @@
   line-height: calc(var(--jp-private-menubar-height) - var(--jp-border-width));
 }
 
+.lm-MenuBar-content:focus-visible {
+  outline-offset: -3px; /* this value is a compromise between Firefox, Chrome,
+    and Safari over this outline's visibility and discretion */
+}
+
+.lm-MenuBar:focus-visible {
+  outline: 1px solid var(--jp-accept-color-active, var(--jp-brand-color1));
+  outline-offset: -1px;
+}
+
+.lm-MenuBar-menu:focus-visible,
+.lm-MenuBar-item:focus-visible,
+.lm-Menu-item:focus-visible {
+  outline: unset;
+  outline-offset: unset;
+  -moz-outline-radius: unset;
+}
+
```

Comment:
With the changes here, the tabbing/focus user experience for the Lumino menubar is not great. The menubar UX needs to be changed at the Lumino level. But my intention with this PR is not to fix all keyboard navigation issues. My intention with this PR is to make focus visible. There are elements that shouldn't receive focus but do, and vice versa. But even if the tabbing order or the set of elements that can receive focus doesn't make sense, that's not a reason to hide the focus indicator; it's a reason to fix which elements receive focus and in which order. At least that's how I see it.

Reply:
Would this changes be applicable to lumino too therefore be better included as default lumino styles?

> The menubar UX needs to be changed at the Lumino level.

Does https://github.com/jupyterlab/lumino/pull/465 address the changes needed? Would you be able to help reviewing it?

Reply:
Oh neat! I'll take a look at that PR today, thanks @krassowski!

The problem with the Lumino menubar is that it renders an HTML structure like this:

```html
<div tabindex="0">
  <div tabindex="0">
    <ul>
      <li tabindex="0">
```

So the user has to tab through an outer container, then an inner container, and then finally each item in the menu bar (File, Edit, etc.). I think that ideally the menubar component should output a single DOM node with tabindex=0 so that the end user can navigate to the menubar, and then from there use arrow keys to access the different parts of the menubar. I'm hoping that's what jupyterlab/lumino#465 does.

Reply:
I like the idea of simplifying the DOM output of menubar as it could help with performance too.